src/credential/issuance/03-start-credential-issuance.ts

```diff
@@ -0,0 +1,336 @@
+    },
+    body: authorizationRequestFormBody.toString(),
+  })
+    .then(hasStatus(200))
+    .then((res) => res.json())
+    .then((body) => TokenResponse.safeParse(body));
+
+  if (!tokenRes.success) {
+    throw new ValidationFailed(tokenRes.error.message);
+  }
+
+  /**
+   * Validates the token response and extracts the access token, c_nonce and c_nonce_expires_in.
+   */
+  const accessTokenResponse = tokenRes.data;
+  const credentialUrl = issuerConf.openid_credential_issuer.credential_endpoint;
+
+  /**
+   * JWT proof token to bind the request nonce to the key that will bind the holder User with the Credential
+   * This is presented along with the access token to the Credential Endpoint as proof of possession of the private key used to sign the Access Token.
+   * @see https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#name-proof-types
+   */
+  const signedNonceProof = await createNonceProof(
+    accessTokenResponse.c_nonce,
+    clientId,
+    credentialUrl,
+    credentialCryptoContext
+  );
+
+  // Validation of accessTokenResponse.authorization_details if contain credentialDefinition
+  const constainsCredentialDefinition =
+    accessTokenResponse.authorization_details.some(
+      (c) =>
+        c.credential_configuration_id ===
+          credentialDefinition.credential_configuration_id &&
+        c.format === credentialDefinition.format &&
+        c.type === credentialDefinition.type
+    );
+
+  if (!constainsCredentialDefinition) {
+    throw new ValidationFailed(
+      "The access token response does not contain the requested credential"
+    );
+  }
+
+  /** The credential request body */
+  const credentialRequestFormBody = {
+    credential_definition: {
+      type: [credentialDefinition.credential_configuration_id],
+    },
+    format: credentialDefinition.format,
+    proof: {
+      jwt: signedNonceProof,
+      proof_type: "jwt",
+    },
+  };
+
+  const credentialRes = await appFetch(credentialUrl, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+      DPoP: tokenRequestSignedDPop,
+      Authorization: `${accessTokenResponse.token_type} ${accessTokenResponse.access_token}`,
+    },
+    body: JSON.stringify(credentialRequestFormBody),
+  })
+    .then(hasStatus(200))
+    .then((res) => res.json())
+    .then((body) => CredentialResponse.safeParse(body));
+
+  if (!credentialRes.success) {
+    throw new ValidationFailed(credentialRes.error.message);
+  }
+
+  return credentialRes.data;
+};
+
+export const createNonceProof = async (
+  nonce: string,
+  issuer: string,
+  audience: string,
+  ctx: CryptoContext
+): Promise<string> => {
+  const jwk = await ctx.getPublicKey();
+  return new SignJWT(ctx)
+    .setPayload({
+      nonce,
+    })
+    .setProtectedHeader({
+      typ: "openid4vci-proof+jwt",
+      jwk,
+    })
+    .setAudience(audience)
+    .setIssuer(issuer)
+    .setIssuedAt()
+    .setExpirationTime("1h")
```

Comment:
```suggestion
    .setExpirationTime("5min")
```

Reply:
Fixed in 5621aeba7e2bece3d95a5060353a4c72e7eec56b, thanks.